libraries/standard/mqtt/src/mqtt.c

```diff
@@ -163,7 +163,32 @@ MQTTStatus_t MQTT_Unsubscribe( MQTTContext_t * const pContext,
 
 MQTTStatus_t MQTT_Disconnect( MQTTContext_t * const pContext )
 {
-    return MQTTSuccess;
+    size_t packetSize;
+    int32_t bytesSent;
+
+    MQTTStatus_t status = MQTT_GetDisconnectPacketSize( &packetSize );
+
+    if( status == MQTTSuccess )
+    {
+        status = MQTT_SerializeDisconnect( &( pContext->networkBuffer ) );
```

Comment:
The application needs to know the size of the Disconnect packet before passing in its buffer to ensure the buffer is big enough.  So the call to get the size of the Disconnect packet is a separate function in case the application wants to malloc the buffer for it.

Reply:
This is the case where high level API is using serializer (low level) API.  If an application directly uses the serilizer API, the application may want to know the size of the resulting packet. However in case of disconnect packet,  the packet size is constant, we may want to have it as #define.

Reply:
For consistency with how sizing works for other packets (through a function call), this should also be a function call.

Reply:
networkBuffer is a struct with a size member.